src/Content/Conversation/Repository/UserDefinedChannel.php

```diff
@@ -140,40 +150,177 @@ public function save(Entity\UserDefinedChannel $Channel): Entity\UserDefinedChan
 		return $Channel;
 	}
 
+	private function isValid(string $searchtext): bool
+	{
+		if ($searchtext == '') {
+			return true;
+		}
+
+		$this->db->insert('check-full-text-search', ['pid' => getmypid(), 'searchtext' => $searchtext], Database::INSERT_UPDATE);
+		$result = $this->db->select('check-full-text-search', [], ["`pid` = ? AND MATCH (`searchtext`) AGAINST (? IN BOOLEAN MODE)", getmypid(), $this->escapeKeywords($searchtext)]);
+		$this->db->delete('check-full-text-search', ['pid' => getmypid()]);
```

Comment:
What are the insertion and deletion for?

Reply:
We need some data to check this type of search.

Reply:
I don't understand:
- Data is inserted
- Data is selected
- Data is deleted

What does this tell us?

Reply:
`$result` is important. It will be `false` when the syntax of the search is invalid. This can happen for example when you use control characters like `-`, `+`, `(` or `)` in a wrong way.

Reply:
Ah, you're just testing the syntax? You can do it in a single query like this:

```
SELECT MATCH (`field`) AGAINST (? IN BOOLEAN MODE) FROM `table` LIMIT 1;
```
No need for insertion nor deletion.

Reply:
I'm not sure, if the system reacts differently, when there is no data in the target table.

Reply:
Then pick a table where we expect data!

Reply:
I already changed the code.

Reply:
Does it work as expected even without data?

Reply:
I tested that yesterday, then I made the changes.